File: frontend/src/types/library.ts
export type LibraryPartItem = {
  tag: string
  file_name: string
  letter_url: string
  a4_url: string
}

export type LibraryItem = {
  partset_id: string
  private_id: string | null
  score_id: string | null
  title: string | null
  composer: string | null
  publisher: string | null
  admin: boolean
  parts_ready: boolean
  parts: LibraryPartItem[]
  score_pdf_url: string | null
}

export type LibraryResponse = {
  items: LibraryItem[]
}

File: api/app/schemas/library.py
from pydantic import BaseModel


class LibraryPartItem(BaseModel):
    tag: str
    file_name: str
    letter_url: str
    a4_url: str


class LibraryItem(BaseModel):
    partset_id: str
    private_id: str | None = None
    score_id: str | None = None
    title: str | None = None
    composer: str | None = None
    publisher: str | None = None
    admin: bool
    parts_ready: bool
    parts: list[LibraryPartItem]
    score_pdf_url: str | None = None
    imslp_id: str | None = None


class LibraryResponse(BaseModel):
    items: list[LibraryItem]


class FavoriteStatusResponse(BaseModel):
    favorite: bool


class FavoriteActionRequest(BaseModel):
    action: str
